src/index.ts

```diff
@@ -180,15 +202,24 @@ class ClientApp {
                 this.assertNonEmptyString(cfg.pcOrigin, 'pcOrigin');
                 this._customPcOrigin = cfg.pcOrigin;
             }
+
+            if("gcTargetEnvQueryParam" in cfg){
+                const paramName = cfg.gcTargetEnvQueryParam;
+                this.assertNonEmptyString(paramName, 'target env query param name');
+                const parsedQueryString = queryString.parse(ClientApp._getQueryString() || '');
+                const paramValue = decodeURI(parsedQueryString[paramName]);
+                this.gcTargetEnv = paramValue;
+            }
```

Comment:
ultimately, i think we need to parse/acquire this soup of values and then build up a PcEnv representation.  We might want to add environment to that type in addition to the hostname.  We should be able to map envs to tlds via sd-web.  If we consistently set the internal `_pcEnv` prop, then the `pcEnvironment` will still be supported.

The other option is that we could rev the major version and just support host and target env.  Let's chat about this tomorrow.